utils/filesystem/zip.go

```diff
@@ -223,8 +234,10 @@ func (fs *VFS) unzip(ctx context.Context, source string, destination string, lim
 			return fileList, subErr
 		}
 
-		// Keep list of files unzipped
-		fileList = append(fileList, filePath)
+		// Keep list of files unzipped (except zip files as they will be handled later)
+		if filepath.Ext(zippedFile.Name) != zipExt {
```

Comment:
the problem is that zip does not need to be with .zip, it could be .pack, .gz, .7zip ....

could you add a function `isZip(file string)` and use it here